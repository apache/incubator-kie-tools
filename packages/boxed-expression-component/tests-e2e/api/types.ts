import {
  DMN15__tConditional,
  DMN15__tContext,
  DMN15__tDecisionTable,
  DMN15__tFilter,
  DMN15__tFor,
  DMN15__tFunctionDefinition,
  DMN15__tInvocation,
  DMN15__tList,
  DMN15__tLiteralExpression,
  DMN15__tQuantified,
  DMN15__tRelation,
} from "@kie-tools/dmn-marshaller/dist/schemas/dmn-1_5/ts-gen/types";

export type BoxedLiteral = { __$$element: "literalExpression" } & DMN15__tLiteralExpression;
export type BoxedRelation = { __$$element: "relation" } & DMN15__tRelation;
export type BoxedContext = { __$$element: "context" } & DMN15__tContext;
export type BoxedDecisionTable = { __$$element: "decisionTable" } & DMN15__tDecisionTable;
export type BoxedList = { __$$element: "list" } & DMN15__tList;
export type BoxedInvocation = { __$$element: "invocation" } & DMN15__tInvocation;
export type BoxedFunction = { __$$element: "functionDefinition" } & DMN15__tFunctionDefinition;
export type BoxedFor = { __$$element: "for" } & DMN15__tFor;
export type BoxedEvery = { __$$element: "every" } & DMN15__tQuantified;
export type BoxedSome = { __$$element: "some" } & DMN15__tQuantified;
export type BoxedConditional = { __$$element: "conditional" } & DMN15__tConditional;
export type BoxedFilter = { __$$element: "filter" } & DMN15__tFilter;

export type Normalized<T> = WithRequiredDeep<T, "@_id">;

type WithRequiredDeep<T, K extends keyof any> = T extends undefined
  ? T
  : T extends Array<infer U>
    ? Array<WithRequiredDeep<U, K>>
    : { [P in keyof T]: WithRequiredDeep<T[P], K> } & (K extends keyof T
        ? { [P in K]-?: NonNullable<WithRequiredDeep<T[P], K>> }
        : T);
