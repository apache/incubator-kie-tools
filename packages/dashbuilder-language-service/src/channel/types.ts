/*
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { TextDocument } from "vscode-languageserver-textdocument";
import { CompletionItemKind, Range } from "vscode-languageserver-types";

export declare type DashbuilderLsNodeType = "object" | "array" | "property" | "string" | "number" | "boolean" | "null";
export declare type DashbuilderJsonPath = (string | number)[];

/**
 * The AST node used in the LanguageServices
 */
export type DashbuilderLsNode = {
  type: DashbuilderLsNodeType;
  value?: any;
  offset: number;
  length: number;
  colonOffset?: number;
  parent?: DashbuilderLsNode;
  children?: DashbuilderLsNode[];
};

export interface TranslateArgs {
  completion: object | string;
  completionItemKind: CompletionItemKind;
  currentNodeRange?: Range;
  cursorOffset: number;
  document: TextDocument;
  overwriteRange?: Range;
}
