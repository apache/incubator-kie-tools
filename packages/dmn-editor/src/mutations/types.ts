import { NodeType } from "../diagram/connections/graphStructure";
import { NODE_TYPES } from "../diagram/nodes/NodeTypes";

export enum NodeNature {
  DRG_ELEMENT = "DRG_ELEMENT",
  ARTIFACT = "ARTIFACT",
}

export const nodeNatures: Record<NodeType, NodeNature> = {
  [NODE_TYPES.inputData]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.decision]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.bkm]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.knowledgeSource]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.decisionService]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.textAnnotation]: NodeNature.ARTIFACT,
  [NODE_TYPES.group]: NodeNature.ARTIFACT,
};
