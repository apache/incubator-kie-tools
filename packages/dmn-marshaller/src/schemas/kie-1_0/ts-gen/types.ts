// This file was automatically generated

import { XmlParserTsRootElementBaseType } from "@kie-tools/xml-parser-ts";

export type KIE__tComponentsWidthsExtension = {
  ComponentWidths?: KIE__tComponentWidths[]; // from type KIE__tComponentsWidthsExtension @ KIE.xsd
};

export type KIE__tComponentWidths = {
  "@_dmnElementRef"?: string; // from type KIE__tComponentWidths @ KIE.xsd
  width?: number[]; // from type KIE__tComponentWidths @ KIE.xsd
};

export type KIE__tAttachment = {
  "@_url"?: string; // from type KIE__tAttachment @ KIE.xsd
  "@_name"?: string; // from type KIE__tAttachment @ KIE.xsd
};
