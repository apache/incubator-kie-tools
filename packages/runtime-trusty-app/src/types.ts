/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
export interface TrustyContextValue {
  config: {
    counterfactualEnabled: boolean;
    explanationEnabled: boolean;
    serverRoot: string;
    basePath: string;
    useHrefLinks: boolean;
  };
}

export type RemoteData<E, D> =
  | { status: RemoteDataStatus.NOT_ASKED }
  | { status: RemoteDataStatus.LOADING }
  | { status: RemoteDataStatus.FAILURE; error: E }
  | { status: RemoteDataStatus.SUCCESS; data: D };

export enum RemoteDataStatus {
  NOT_ASKED,
  LOADING,
  FAILURE,
  SUCCESS,
}

export interface Execution {
  executionId: string;
  executionDate: string;
  executedModelName: string;
  executionType: string;
  executionSucceeded: boolean;
  executorName: string;
}

export interface Executions {
  headers: Execution[];
  total: number;
}

export interface ExecutionRouteParams {
  executionId: string;
  executionType: string;
}

export interface ItemObject {
  name: string;
  value: ItemObjectValue;
}

export type ItemObjectValue = ItemObjectUnit | ItemObjectCollection | ItemObjectStructure;

export interface ItemObjectUnit {
  kind: "UNIT";
  type: string;
  value: string | number | boolean | Array<string | number | boolean> | null;
}

export interface ItemObjectCollection {
  kind: "COLLECTION";
  type: string;
  value: ItemObjectValue[] | null;
}

export type ItemObjectMap = { [key: string]: ItemObjectValue };

export interface ItemObjectStructure {
  kind: "STRUCTURE";
  type: string;
  value: ItemObjectMap;
}

export interface InputRow {
  inputLabel: string;
  inputValue?: ItemObject["value"];
  key: string;
  category: string;
}

export enum evaluationStatus {
  EVALUATING = "Evaluating",
  FAILED = "Failed",
  NOT_EVALUATED = "Not evaluated",
  SKIPPED = "Skipped",
  SUCCEEDED = "Succeeded",
}

export type evaluationStatusStrings = keyof typeof evaluationStatus;

export interface Outcome {
  outcomeId: string;
  outcomeName: string;
  outcomeResult: ItemObjectValue;
  evaluationStatus: evaluationStatusStrings;
  hasErrors: boolean;
  messages: string[];
}

export interface FeatureScores {
  featureName: string;
  featureScore: number;
}

export interface Saliency {
  outcomeId: string;
  featureImportance: FeatureScores[];
}

export enum SaliencyStatus {
  SUCCEEDED = "SUCCEEDED",
  FAILED = "FAILED",
}

export type SaliencyStatusStrings = keyof typeof SaliencyStatus;

export interface Saliencies {
  status: SaliencyStatusStrings;
  statusDetail: string;
  saliencies: Saliency[];
}

export interface ServiceIdentifier {
  groupId?: string;
  artifactId?: string;
  version?: string;
}

export interface ModelData {
  deploymentDate?: string;
  modelId?: string;
  name: string;
  namespace: string;
  modelVersion: string;
  dmnVersion: string;
  serviceIdentifier: ServiceIdentifier;
  model: string;
}

export interface CFSearchInput {
  name: string;
  value: CFSearchInputValue;
}

export type CFSearchInputValue = CFSearchInputUnit | CFSearchInputCollection | CFSearchInputStructure;

export interface CFSearchInputUnit {
  kind: "UNIT";
  type: string;
  fixed?: boolean;
  domain?: CFNumericalDomain | CFCategoricalDomain;
  originalValue: ItemObject["value"];
}

export interface CFNumericalDomain {
  type: "RANGE";
  lowerBound?: number;
  upperBound?: number;
}

export interface CFCategoricalDomain {
  type: "CATEGORICAL";
  categories: string[];
}

export interface CFSearchInputCollection {
  kind: "COLLECTION";
  type: string;
  value: Array<CFSearchInputValue> | null;
}

export type CFSearchInputValueMap = { [key: string]: CFSearchInputValue };

export interface CFSearchInputStructure {
  kind: "STRUCTURE";
  type: string;
  value: CFSearchInputValueMap | null;
}

export enum CFGoalRole {
  UNSUPPORTED,
  ORIGINAL,
  FIXED,
  FLOATING,
}

export interface CFGoal {
  id: string;
  name: string;
  role: CFGoalRole;
  value: ItemObjectValue;
  originalValue: CFGoal["value"];
}

export type CFResult = Array<unknown>;

export interface CFStatus {
  isDisabled: boolean;
  executionStatus: CFExecutionStatus;
  lastExecutionTime: null | string;
}

export enum CFExecutionStatus {
  COMPLETED,
  RUNNING,
  NOT_STARTED,
  FAILED,
  NO_RESULTS,
}

export type CFAnalysisResetType = "NEW" | "EDIT";

export interface CFAnalysisExecution {
  executionId: string;
  counterfactualId: string;
  maxRunningTimeSeconds: number;
}

export interface CFAnalysisResult {
  executionId: string;
  counterfactualId: string;
  type: "counterfactual";
  valid: boolean;
  status: "SUCCEEDED" | "FAILED";
  statusDetails: string;
  solutionId: string;
  isValid: boolean;
  stage: "INTERMEDIATE" | "FINAL";
  inputs: ItemObject[];
  outputs: ItemObject[];
  sequenceId: number;
}

export interface CFAnalysisResultsSets extends CFAnalysisExecution {
  goals: CFGoal[];
  searchDomains: CFSearchInput[];
  solutions: CFAnalysisResult[];
}

export type CFSupportMessage = {
  id: string;
  message: string;
};
