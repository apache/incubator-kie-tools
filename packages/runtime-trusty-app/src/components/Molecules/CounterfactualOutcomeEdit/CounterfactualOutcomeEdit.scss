.counterfactual-outcome__floating {
  padding-top: var(--pf-global--spacer--form-element);
}
