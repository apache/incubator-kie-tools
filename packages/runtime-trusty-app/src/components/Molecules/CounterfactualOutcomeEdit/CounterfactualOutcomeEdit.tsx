/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React, { useEffect, useMemo, useState } from "react";
import {
  Checkbox,
  FormGroup,
  NumberInput,
  Switch,
  Text,
  TextContent,
  TextInput,
  TextVariants,
} from "@patternfly/react-core/dist/js/components";
import { v4 as uuid } from "uuid";
import { CFGoal, CFGoalRole, ItemObjectUnit } from "../../../types";
import CounterfactualOutcomeUnsupported from "../../Atoms/CounterfactualOutcomeUnsupported/CounterfactualOutcomeUnsupported";
import "./CounterfactualOutcomeEdit.scss";

type CounterfactualOutcomeEditProps = {
  goal: CFGoal;
  index: number;
  onUpdateGoal: (goal: CFGoal) => void;
};

const CounterfactualOutcomeEdit = (props: CounterfactualOutcomeEditProps) => {
  const { goal, index, onUpdateGoal } = props;

  if (goal.role === CFGoalRole.UNSUPPORTED) {
    return <CounterfactualOutcomeUnsupported goal={goal} />;
  }

  let valueEdit;
  switch (typeof goal.value.value) {
    case "boolean":
      valueEdit = <CounterfactualOutcomeBoolean goal={goal} index={index} onUpdateGoal={onUpdateGoal} />;
      break;
    case "number":
      valueEdit = <CounterfactualOutcomeNumber goal={goal} index={index} onUpdateGoal={onUpdateGoal} />;
      break;
    case "string":
      valueEdit = <CounterfactualOutcomeString goal={goal} index={index} onUpdateGoal={onUpdateGoal} />;
      break;
  }

  return (
    <>
      <FormGroup fieldId={goal.id} label={goal.name}>
        {valueEdit}
      </FormGroup>
      <Checkbox
        id={goal.id}
        className="counterfactual-outcome__floating"
        isChecked={goal.role === CFGoalRole.FLOATING}
        onChange={(checked) => {
          let updatedRole = CFGoalRole.FIXED;
          if (goal.value === goal.originalValue) {
            updatedRole = CFGoalRole.ORIGINAL;
          }
          if (checked) {
            updatedRole = CFGoalRole.FLOATING;
          }
          onUpdateGoal({
            ...goal,
            role: updatedRole,
          });
        }}
        label={
          <TextContent>
            <Text component={TextVariants.small}>Automatically adjust for counterfactual.</Text>
          </TextContent>
        }
      />
    </>
  );
};

export default CounterfactualOutcomeEdit;

const CounterfactualOutcomeBoolean = (props: CounterfactualOutcomeEditProps) => {
  const { goal, onUpdateGoal } = props;
  const [booleanValue, setBooleanValue] = useState(goal.value.value as boolean);

  useEffect(() => {
    setBooleanValue(goal.value.value as boolean);
  }, [goal.value]);

  const handleChange = (checked: boolean) => {
    if (isFloating(goal)) {
      return;
    }
    onUpdateGoal({
      ...goal,
      value: { ...(goal.value as ItemObjectUnit), value: checked },
    });
  };

  return (
    <Switch
      id={uuid()}
      label="True"
      labelOff="False"
      isChecked={booleanValue}
      isDisabled={isFloating(goal)}
      onChange={handleChange}
    />
  );
};

const CounterfactualOutcomeNumber = (props: CounterfactualOutcomeEditProps) => {
  const { goal, index, onUpdateGoal } = props;
  const [numberValue, setNumberValue] = useState<number>();

  const decimalPlaces = goal.value.value.toString().split(".")[1]?.length || 0;

  const touchSpinWidth = useMemo(() => String(goal.value).length + 2, [goal.value]);

  const onMinus = () => {
    if (isFloating(goal)) {
      return;
    }
    onUpdateGoal({
      ...goal,
      value: {
        ...(goal.value as ItemObjectUnit),
        value: Number(((goal.value.value as number) - 1).toFixed(decimalPlaces)),
      },
    });
  };

  const onChange = (event) => {
    if (isFloating(goal)) {
      return;
    }
    onUpdateGoal({
      ...goal,
      value: {
        ...(goal.value as ItemObjectUnit),
        value: Number(event.target.value),
      },
    });
  };

  const onPlus = () => {
    if (isFloating(goal)) {
      return;
    }
    onUpdateGoal({
      ...goal,
      value: {
        ...(goal.value as ItemObjectUnit),
        value: Number(((goal.value.value as number) + 1).toFixed(decimalPlaces)),
      },
    });
  };

  useEffect(() => {
    setNumberValue(goal.value.value as number);
  }, [goal.value]);

  return (
    <NumberInput
      value={numberValue}
      onMinus={onMinus}
      onChange={onChange}
      onPlus={onPlus}
      inputName={`goal-${index}-${goal.name}`}
      id={`goal-${index}-${goal.name}`}
      inputAriaLabel="`${outcome.outcomeName} input`"
      minusBtnAriaLabel="minus"
      plusBtnAriaLabel="plus"
      widthChars={touchSpinWidth}
      isDisabled={isFloating(goal)}
    />
  );
};

const CounterfactualOutcomeString = (props: CounterfactualOutcomeEditProps) => {
  const { goal, index, onUpdateGoal } = props;

  const handleChange = (value: string) => {
    if (isFloating(goal)) {
      return;
    }
    onUpdateGoal({
      ...goal,
      value: { ...(goal.value as ItemObjectUnit), value: value },
    });
  };

  return (
    <TextInput
      id={`goal-${index}-${goal.name}`}
      name={`goal-${index}-${goal.name}`}
      value={goal.value.value as string}
      onChange={handleChange}
      style={{ width: 250 }}
      isDisabled={isFloating(goal)}
    />
  );
};

const isFloating = (goal: CFGoal) => {
  return goal.role === CFGoalRole.FLOATING;
};
