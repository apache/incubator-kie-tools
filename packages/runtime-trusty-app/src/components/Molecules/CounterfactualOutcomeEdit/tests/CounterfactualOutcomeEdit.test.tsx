/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from "react";
import { mount } from "enzyme";
import { CFGoal, CFGoalRole } from "../../../../types";
import CounterfactualOutcomeEdit from "../CounterfactualOutcomeEdit";

const onUpdateGoal = jest.fn();

describe("CounterfactualOutcomeEdit", () => {
  test("renders an number input for numerical goals", () => {
    const wrapper = mount(<CounterfactualOutcomeEdit goal={goals[0]} index={0} onUpdateGoal={onUpdateGoal} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("updates a goal value using the minus stepper button", () => {
    const wrapper = mount(<CounterfactualOutcomeEdit goal={goals[0]} index={0} onUpdateGoal={onUpdateGoal} />);

    wrapper.find('CounterfactualOutcomeEdit ButtonBase [aria-label="minus"]').simulate("click");

    expect(onUpdateGoal).toBeCalledWith({
      ...goals[0],
      value: {
        ...goals[0].value,
        value: 0.65,
      },
    });
  });
  test("updates a goal value using the plus stepper button", () => {
    const wrapper = mount(<CounterfactualOutcomeEdit goal={goals[1]} index={1} onUpdateGoal={onUpdateGoal} />);

    wrapper.find('CounterfactualOutcomeEdit ButtonBase [aria-label="plus"]').simulate("click");

    expect(onUpdateGoal).toBeCalledWith({
      ...goals[1],
      value: {
        ...goals[1].value,
        value: 2.03,
      },
    });
  });
  test("updates a goal with an integer value", () => {
    const wrapper = mount(<CounterfactualOutcomeEdit goal={goals[2]} index={2} onUpdateGoal={onUpdateGoal} />);

    wrapper.find('CounterfactualOutcomeEdit ButtonBase [aria-label="plus"]').simulate("click");

    expect(onUpdateGoal).toBeCalledWith({
      ...goals[2],
      value: {
        ...goals[2].value,
        value: 6,
      },
    });
  });
});

const goals: CFGoal[] = [
  {
    id: "goal1",
    name: "field1",
    role: CFGoalRole.ORIGINAL,
    value: {
      kind: "UNIT",
      type: "number",
      value: 1.65,
    },
    originalValue: {
      kind: "UNIT",
      type: "number",
      value: 1.65,
    },
  },
  {
    id: "goal1",
    name: "field1",
    role: CFGoalRole.ORIGINAL,
    value: {
      kind: "UNIT",
      type: "number",
      value: 1.03,
    },
    originalValue: {
      kind: "UNIT",
      type: "number",
      value: 1.03,
    },
  },
  {
    id: "goal3",
    name: "field3",
    role: CFGoalRole.ORIGINAL,
    value: {
      kind: "UNIT",
      type: "number",
      value: 5,
    },
    originalValue: {
      kind: "UNIT",
      type: "number",
      value: 5,
    },
  },
];
