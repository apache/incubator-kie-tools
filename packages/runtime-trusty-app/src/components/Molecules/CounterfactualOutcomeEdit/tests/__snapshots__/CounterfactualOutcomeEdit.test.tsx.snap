// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterfactualOutcomeEdit renders an number input for numerical goals 1`] = `
<CounterfactualOutcomeEdit
  goal={
    Object {
      "id": "goal1",
      "name": "field1",
      "originalValue": Object {
        "kind": "UNIT",
        "type": "number",
        "value": 1.65,
      },
      "role": 1,
      "value": Object {
        "kind": "UNIT",
        "type": "number",
        "value": 1.65,
      },
    }
  }
  index={0}
  onUpdateGoal={[MockFunction]}
>
  <FormGroup
    fieldId="goal1"
    label="field1"
  >
    <GenerateId
      prefix="pf-random-id-"
    >
      <div
        className="pf-c-form__group"
      >
        <div
          className="pf-c-form__group-label"
        >
          <label
            className="pf-c-form__label"
            htmlFor="goal1"
          >
            <span
              className="pf-c-form__label-text"
            >
              field1
            </span>
          </label>
           
        </div>
        <div
          className="pf-c-form__group-control"
        >
          <CounterfactualOutcomeNumber
            goal={
              Object {
                "id": "goal1",
                "name": "field1",
                "originalValue": Object {
                  "kind": "UNIT",
                  "type": "number",
                  "value": 1.65,
                },
                "role": 1,
                "value": Object {
                  "kind": "UNIT",
                  "type": "number",
                  "value": 1.65,
                },
              }
            }
            index={0}
            onUpdateGoal={[MockFunction]}
          >
            <NumberInput
              id="goal-0-field1"
              inputAriaLabel="\`\${outcome.outcomeName} input\`"
              inputName="goal-0-field1"
              isDisabled={false}
              minusBtnAriaLabel="minus"
              onChange={[Function]}
              onMinus={[Function]}
              onPlus={[Function]}
              plusBtnAriaLabel="plus"
              value={1.65}
              widthChars={17}
            >
              <div
                className="pf-c-number-input"
                id="goal-0-field1"
                style={
                  Object {
                    "--pf-c-number-input--c-form-control--width-chars": 17,
                  }
                }
              >
                <InputGroup>
                  <div
                    className="pf-c-input-group"
                  >
                    <Button
                      aria-label="minus"
                      isDisabled={false}
                      onClick={[Function]}
                      variant="control"
                    >
                      <ButtonBase
                        aria-label="minus"
                        innerRef={null}
                        isDisabled={false}
                        onClick={[Function]}
                        variant="control"
                      >
                        <button
                          aria-disabled={false}
                          aria-label="minus"
                          className="pf-c-button pf-m-control"
                          data-ouia-component-id="OUIA-Generated-Button-control-1"
                          data-ouia-component-type="PF4/Button"
                          data-ouia-safe={true}
                          disabled={false}
                          onClick={[Function]}
                          role={null}
                          type="button"
                        >
                          <span
                            className="pf-c-number-input__icon"
                          >
                            <MinusIcon
                              aria-hidden="true"
                              color="currentColor"
                              noVerticalAlign={false}
                              size="sm"
                            >
                              <svg
                                aria-hidden="true"
                                aria-labelledby={null}
                                fill="currentColor"
                                height="1em"
                                role="img"
                                style={
                                  Object {
                                    "verticalAlign": "-0.125em",
                                  }
                                }
                                viewBox="0 0 448 512"
                                width="1em"
                              >
                                <path
                                  d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                                />
                              </svg>
                            </MinusIcon>
                          </span>
                        </button>
                      </ButtonBase>
                    </Button>
                    <TextInput
                      aria-label="\`\${outcome.outcomeName} input\`"
                      name="goal-0-field1"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      type="number"
                      validated="default"
                      value={1.65}
                    >
                      <TextInputBase
                        aria-label="\`\${outcome.outcomeName} input\`"
                        className=""
                        innerRef={null}
                        isDisabled={false}
                        isIconSprite={false}
                        isLeftTruncated={false}
                        isReadOnly={false}
                        isRequired={false}
                        name="goal-0-field1"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onKeyDown={[Function]}
                        ouiaSafe={true}
                        type="number"
                        validated="default"
                        value={1.65}
                      >
                        <input
                          aria-invalid={false}
                          aria-label="\`\${outcome.outcomeName} input\`"
                          className="pf-c-form-control"
                          data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                          data-ouia-component-type="PF4/TextInput"
                          data-ouia-safe={true}
                          disabled={false}
                          name="goal-0-field1"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          required={false}
                          type="number"
                          value={1.65}
                        />
                      </TextInputBase>
                    </TextInput>
                    <Button
                      aria-label="plus"
                      isDisabled={false}
                      onClick={[Function]}
                      variant="control"
                    >
                      <ButtonBase
                        aria-label="plus"
                        innerRef={null}
                        isDisabled={false}
                        onClick={[Function]}
                        variant="control"
                      >
                        <button
                          aria-disabled={false}
                          aria-label="plus"
                          className="pf-c-button pf-m-control"
                          data-ouia-component-id="OUIA-Generated-Button-control-2"
                          data-ouia-component-type="PF4/Button"
                          data-ouia-safe={true}
                          disabled={false}
                          onClick={[Function]}
                          role={null}
                          type="button"
                        >
                          <span
                            className="pf-c-number-input__icon"
                          >
                            <PlusIcon
                              aria-hidden="true"
                              color="currentColor"
                              noVerticalAlign={false}
                              size="sm"
                            >
                              <svg
                                aria-hidden="true"
                                aria-labelledby={null}
                                fill="currentColor"
                                height="1em"
                                role="img"
                                style={
                                  Object {
                                    "verticalAlign": "-0.125em",
                                  }
                                }
                                viewBox="0 0 448 512"
                                width="1em"
                              >
                                <path
                                  d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                                />
                              </svg>
                            </PlusIcon>
                          </span>
                        </button>
                      </ButtonBase>
                    </Button>
                  </div>
                </InputGroup>
              </div>
            </NumberInput>
          </CounterfactualOutcomeNumber>
        </div>
      </div>
    </GenerateId>
  </FormGroup>
  <Checkbox
    className="counterfactual-outcome__floating"
    component="div"
    id="goal1"
    isChecked={false}
    isDisabled={false}
    isRequired={false}
    isValid={true}
    label={
      <TextContent>
        <Text
          component="small"
        >
          Automatically adjust for counterfactual.
        </Text>
      </TextContent>
    }
    onChange={[Function]}
    ouiaSafe={true}
  >
    <div
      className="pf-c-check counterfactual-outcome__floating"
    >
      <input
        aria-invalid={false}
        checked={false}
        className="pf-c-check__input"
        data-ouia-component-id="OUIA-Generated-Checkbox-1"
        data-ouia-component-type="PF4/Checkbox"
        data-ouia-safe={true}
        disabled={false}
        id="goal1"
        onChange={[Function]}
        required={false}
        type="checkbox"
      />
      <label
        className="pf-c-check__label"
        htmlFor="goal1"
      >
        <TextContent>
          <div
            className="pf-c-content"
          >
            <Text
              component="small"
            >
              <small
                className=""
                data-ouia-component-id="OUIA-Generated-Text-1"
                data-ouia-component-type="PF4/Text"
                data-ouia-safe={true}
                data-pf-content={true}
              >
                Automatically adjust for counterfactual.
              </small>
            </Text>
          </div>
        </TextContent>
      </label>
    </div>
  </Checkbox>
</CounterfactualOutcomeEdit>
`;
